internal/service/pushbasedlogexport/model_test.go

```diff
@@ -0,0 +1,149 @@
+	testPrefixPath  = "prefix/path"
+	prefixPathEmpty = ""
+	testProjectID   = "661fe3ad234b02027dabcabc"
+)
+
+type sdkToTFModelTestCase struct {
+	apiResp         *admin.PushBasedLogExportProject
+	timeout         *timeouts.Value
+	expectedTFModel *pushbasedlogexport.TFPushBasedLogExportRSModel
+	name            string
+	projectID       string
+}
+
+func TestNewTFPushBasedLogExport(t *testing.T) {
+	currentTime := time.Now()
+
+	testCases := []sdkToTFModelTestCase{
+		{
+			name:      "Complete API response",
+			projectID: testProjectID,
+			apiResp: &admin.PushBasedLogExportProject{
+				BucketName: admin.PtrString(testBucketName),
+				CreateDate: admin.PtrTime(currentTime),
+				IamRoleId:  admin.PtrString(testIAMRoleID),
+				PrefixPath: admin.PtrString(testPrefixPath),
+				State:      admin.PtrString(activeState),
+			},
+			expectedTFModel: &pushbasedlogexport.TFPushBasedLogExportRSModel{
+				ProjectID:  types.StringValue(testProjectID),
+				BucketName: types.StringValue(testBucketName),
+				IamRoleID:  types.StringValue(testIAMRoleID),
+				PrefixPath: types.StringValue(testPrefixPath),
+				State:      types.StringValue(activeState),
+				CreateDate: types.StringPointerValue(conversion.TimePtrToStringPtr(&currentTime)),
+			},
+		},
+		{
+			name:      "Complete API response with empty prefix path",
+			projectID: testProjectID,
+			apiResp: &admin.PushBasedLogExportProject{
+				BucketName: admin.PtrString(testBucketName),
+				CreateDate: admin.PtrTime(currentTime),
+				IamRoleId:  admin.PtrString(testIAMRoleID),
+				PrefixPath: admin.PtrString(prefixPathEmpty),
+				State:      admin.PtrString(activeState),
+			},
+			expectedTFModel: &pushbasedlogexport.TFPushBasedLogExportRSModel{
+				ProjectID:  types.StringValue(testProjectID),
+				BucketName: types.StringValue(testBucketName),
+				IamRoleID:  types.StringValue(testIAMRoleID),
+				PrefixPath: types.StringValue(prefixPathEmpty),
+				State:      types.StringValue(activeState),
+				CreateDate: types.StringPointerValue(conversion.TimePtrToStringPtr(&currentTime)),
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resultModel, _ := pushbasedlogexport.NewTFPushBasedLogExport(context.Background(), tc.projectID, tc.apiResp, tc.timeout)
+			if !reflect.DeepEqual(resultModel, tc.expectedTFModel) {
+				t.Errorf("result model does not match expected output: expected %+v, got %+v", tc.expectedTFModel, resultModel)
+			}
+		})
+	}
+}
+
+type pushBasedLogExportReqTestCase struct {
+	input             *pushbasedlogexport.TFPushBasedLogExportRSModel
+	expectedCreateReq *admin.CreatePushBasedLogExportProjectRequest
+	expectedUpdateReq *admin.PushBasedLogExportProject
+	name              string
+}
+
+func TestNewPushBasedLogExportReq(t *testing.T) {
+	testCases := []pushBasedLogExportReqTestCase{
+		{
+			name: "Valid TF state",
+			input: &pushbasedlogexport.TFPushBasedLogExportRSModel{
+				BucketName: types.StringValue(testBucketName),
```

Comment:
to simplify test we might want to have in testBucketName the value of types.StringValue(testBucketName) so we can have just: BucketName: testBucketName,

same for all